scripts/importer.py

```diff
@@ -18,60 +18,173 @@
+    if not bookmark_types:
+        bookmark_types = ['bookmark', 'keyword']
+    if not output_format:
+        output_format = 'quickmark'
+    if not input_format:
+        input_format = browser_default_input_format[args.browser]
+
+    import_function = {'netscape': import_netscape_bookmarks}
+    import_function[input_format](args.bookmarks, bookmark_types,
+                                  output_format)
 
 
 def get_args():
     """Get the argparse parser."""
     parser = argparse.ArgumentParser(
         epilog="To import bookmarks from Chromium, Firefox or IE, "
-               "export them to HTML in your browsers bookmark manager. "
-               "By default, this script will output in a quickmarks format.")
-    parser.add_argument('browser', help="Which browser? (chromium, firefox)",
-                        choices=['chromium', 'firefox', 'ie'],
-                        metavar='browser')
-    parser.add_argument('-b', help="Output in bookmark format.",
-                        dest='bookmark_format', action='store_true',
-                        default=False, required=False)
+        "export them to HTML in your browsers bookmark manager. ")
+    parser.add_argument(
+        'browser',
+        help="Which browser? {%(choices)s}",
+        choices=browser_default_input_format.keys(),
+        nargs='?',
+        metavar='browser')
+    parser.add_argument(
+        '-i',
+        '--input-format',
+        help='Which input format? (overrides browser default)',
+        choices=set(browser_default_input_format.values()),
+        required=False)
+    parser.add_argument(
+        '-b',
+        help="Output in bookmark format.",
+        dest='bookmark_output',
+        action='store_true',
+        default=False,
+        required=False)
+    parser.add_argument(
+        '-q',
+        help="Output in quickmark format (default).",
+        dest='quickmark_output',
+        action='store_true',
+        default=False,
+        required=False)
+    parser.add_argument(
+        '-s',
+        help="Output search engine format",
+        dest='search_output',
+        action='store_true',
+        default=False,
+        required=False)
+    parser.add_argument(
+        '--oldconfig',
+        help="Output search engine format for old qutebrowser.conf format",
+        default=False,
+        action='store_true',
+        required=False)
+    parser.add_argument(
+        '-S',
+        help="Import search engines",
+        dest='search_query',
+        action='store_true',
+        default=False,
+        required=False)
+    parser.add_argument(
+        '-B',
+        help="Import plain bookmarks (no keywords)",
+        dest='bookmark_query',
+        action='store_true',
+        default=False,
+        required=False)
+    parser.add_argument(
+        '-K',
+        help="Import keywords (no search)",
+        dest='keyword_query',
+        action='store_true',
+        default=False,
+        required=False)
```

Comment:
I'm really not sure whether those arguments make sense - why would I e.g. want to import bookmarks but output them in search URL format?